src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -180,7 +180,7 @@ export interface V2AzureResourcesApi extends AzureResourcesApiBase {
     /**
      * Show a quick picker of app resources. Set `options.type` to filter the picks.
      */
-    pickResource<TModel>(options?: ResourcePickOptions): vscode.ProviderResult<TModel>
+    pickResource<TModel extends ResourceModelBase>(options?: ResourcePickOptions): Promise<TModel>
```

Comment:
This doesn't need to be on the API at all anymore. Just the TDP.

Reply:
Removed. I will wait until I need to access the TDP in the clients to add that to the API (soon).

Reply:
It can be in a different PR but I think we should just add it ASAP. The Storage extension will be converted out of the gate, so it will need it.